panda/src/audiotraits/fmodAudioManager.cxx

```diff
@@ -732,76 +725,450 @@ audio_3d_set_drop_off_factor(PN_stdfloat factor) {
+    }
+    break;
+  case DSP::DT_compressor:
+    {
+      CompressorDSP *comp_conf = DCAST(CompressorDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_COMPRESSOR_THRESHOLD, comp_conf->get_threshold());
+      dsp->setParameterFloat(FMOD_DSP_COMPRESSOR_RATIO, comp_conf->get_ratio());
+      dsp->setParameterFloat(FMOD_DSP_COMPRESSOR_ATTACK, comp_conf->get_attack());
+      dsp->setParameterFloat(FMOD_DSP_COMPRESSOR_RELEASE, comp_conf->get_release());
+      dsp->setParameterFloat(FMOD_DSP_COMPRESSOR_GAINMAKEUP, comp_conf->get_gainmakeup());
+    }
+    break;
+  case DSP::DT_distortion:
+    {
+      DistortionDSP *dist_conf = DCAST(DistortionDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_DISTORTION_LEVEL, dist_conf->get_level());
+    }
+    break;
+  case DSP::DT_echo:
+    {
+      EchoDSP *echo_conf = DCAST(EchoDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_ECHO_DELAY, echo_conf->get_delay());
+      dsp->setParameterFloat(FMOD_DSP_ECHO_FEEDBACK, echo_conf->get_feedback());
+      dsp->setParameterFloat(FMOD_DSP_ECHO_DRYLEVEL, echo_conf->get_drylevel());
+      dsp->setParameterFloat(FMOD_DSP_ECHO_WETLEVEL, echo_conf->get_wetlevel());
+    }
+    break;
+  case DSP::DT_fader:
+    {
+      FaderDSP *fader_conf = DCAST(FaderDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_FADER_GAIN, fader_conf->get_gain());
+    }
+    break;
+  case DSP::DT_flange:
+    {
+      FlangeDSP *flange_conf = DCAST(FlangeDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_FLANGE_MIX, flange_conf->get_mix());
+      dsp->setParameterFloat(FMOD_DSP_FLANGE_DEPTH, flange_conf->get_depth());
+      dsp->setParameterFloat(FMOD_DSP_FLANGE_RATE, flange_conf->get_rate());
+    }
+    break;
+  case DSP::DT_highpass:
+    {
+      HighpassDSP *hp_conf = DCAST(HighpassDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_HIGHPASS_CUTOFF, hp_conf->get_cutoff());
+      dsp->setParameterFloat(FMOD_DSP_HIGHPASS_RESONANCE, hp_conf->get_resonance());
+    }
+    break;
+  case DSP::DT_limiter:
+    {
+      LimiterDSP *lim_conf = DCAST(LimiterDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_LIMITER_RELEASETIME, lim_conf->get_release_time());
+      dsp->setParameterFloat(FMOD_DSP_LIMITER_CEILING, lim_conf->get_ceiling());
+      dsp->setParameterFloat(FMOD_DSP_LIMITER_MAXIMIZERGAIN, lim_conf->get_maximizer_gain());
+    }
+    break;
+  case DSP::DT_lowpass:
+    {
+      LowpassDSP *lp_conf = DCAST(LowpassDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_LOWPASS_CUTOFF, lp_conf->get_cutoff());
+      dsp->setParameterFloat(FMOD_DSP_LOWPASS_RESONANCE, lp_conf->get_resonance());
+    }
+    break;
+  case DSP::DT_normalize:
+    {
+      NormalizeDSP *norm_conf = DCAST(NormalizeDSP, dsp_conf);
+      dsp->setParameterFloat(FMOD_DSP_NORMALIZE_FADETIME, norm_conf->get_fade_time());
+      dsp->setParameterFloat(FMOD_DSP_NORMALIZE_THRESHHOLD, norm_conf->get_threshold());
```

Comment:
This seems to error out 
```
In file included from panda/src/audiotraits/fmod_audio_composite1.cxx:3:
panda/src/audiotraits/fmodAudioManager.cxx:989:30: error: use of undeclared
      identifier 'FMOD_DSP_NORMALIZE_THRESHHOLD'; did you mean
      'FMOD_DSP_NORMALIZE_THRESHOLD'?
      dsp->setParameterFloat(FMOD_DSP_NORMALIZE_THRESHHOLD, norm_conf->g...
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                             FMOD_DSP_NORMALIZE_THRESHOLD
```
Should be revised to FMOD_DSP_NORMALIZE_THRESHOLD?

Reply:
Is this on the latest version of the FMOD Engine?  If so, it must've been renamed to that, because the version of FMOD I have has FMOD_DSP_NORMALIZE_THRESHHOLD.

Reply:
Yeah must be a latest version thing

Reply:
Can you let me know which version of API you used so I can implement this in my custom panda3d version?